### PR TITLE
Console subsystem redesigned

### PR DESCRIPTION
**Pull request by forGGe**
No more console driver libraries. Two parameters drives console allocation on stm32f4xx platform:
`CONFIG_CONSOLE_DEVICE_NUM` - device number of the console
`CONFIG_CONSOLE_BAUD` - baud rate of the desired device number

### CODE REVIEW

File: platform/stm32f4xx/templates/platform_console.hpp.in [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    static auto constexpr baudrate  = @CONFIG_CONSOLE_BAUD@;
+    static auto constexpr word_len  = USART_WordLength_8b;
+    static auto constexpr stop_bit  = USART_StopBits_1;
+    static auto constexpr parity    = USART_Parity_No;
+    static auto constexpr mode      = USART_Mode_Rx | USART_Mode_Tx;
+    static auto constexpr hw_flow   = USART_HardwareFlowControl_None;
+};
+
+using platform_console = ecl::usart_bus< ecl::usart_device::dev_3 >;
+
```
Comment (vadimol) [inline platform/stm32f4xx/templates/platform_console.hpp.in]: why dev_3?

File: platform/stm32f4xx/bypass_console.cpp [hunk 2]
```diff
@@ -0,0 +1,53 @@
+#include <platform_console.hpp>
+
+namespace ecl
+{
+
+struct bypass_console
+{
+    static void init()
+    {
+        // TODO: Duplicates usart_bus::init(), but it is ok for this moment
```
Comment (vadimol) [inline platform/stm32f4xx/bypass_console.cpp]: could you provide more details about why do we need bypass?

File: platform/stm32f4xx/CMakeLists.txt [hunk 3]
```diff
@@ -56,4 +56,28 @@ endif()
 target_compile_definitions(stm32f4xx PUBLIC
 	-DCONFIG_MAX_ISR_PRIORITY=${CONFIG_MAX_ISR_PRIORITY})
 
+# Pick console driver
+message(STATUS "STM32F4XX: Checking [CONFIG_USE_CONSOLE]...")
+if(${CONFIG_USE_CONSOLE})
+	if(NOT DEFINED CONFIG_CONSOLE_DEVICE_NUM)
+		message(FATAL_ERROR "CONFIG_CONSOLE_DEVICE_NUM must be set in order to use console")
```
Comment (vadimol) [inline platform/stm32f4xx/CMakeLists.txt]: The CONFIG_CONSOLE_DEVICE_NUM is difficult to understand. I believe that we need a better way to address specific usart instance in future.

File: platform/stm32f4xx/bypass_console.cpp (hunk 2 above)
Comment (forGGe) [inline platform/stm32f4xx/bypass_console.cpp]: discussed

**Comment by forGGe** [comment]
PR is implemented. Ready for final review.

**Comment by forGGe** [comment]
Fixes/implements #27

**Pull request merged**